plugins/avro-format/core/src/main/java/org/pentaho/di/trans/steps/avro/AvroToPdiConverter.java

```diff
@@ -402,6 +403,6 @@ private boolean isLegacySchema( Schema schema ) {
   }
 
   private boolean isNotNull( Object avroData ) {
-    return avroData != null && !( avroData instanceof NullNode );
```

Comment:
Isn't the core of the problem that the object avroData **_was never_** an instance of `com.fasterxml.jackson.databind.node.NullNode` so the clause ` !( avroData instanceof NullNode );` would always return true

so really you just need `avroData != null && ( avroData != JsonProperties.NULL_VALUE ) `

Now with the change is the import for `com.fasterxml.jackson.databind.node.NullNode` not used?